admin/src/osv_export.rs

```diff
@@ -0,0 +1,96 @@
+//! Backend for the `osv` subcommand.
+
+use std::path::{Path, PathBuf};
+
+use rustsec::{
+    advisory::Informational,
+    fs,
+    osv::OsvAdvisory,
+    repository::git::{GitModificationTimes, GitPath, Repository},
+    Advisory, Collection,
+};
+
+use crate::{
+    error::{Error, ErrorKind},
+    prelude::*,
+};
+
+/// Lists all versions for a crate and prints info on which ones are affected
+pub struct OsvExporter {
+    /// Loaded git repository
+    repository: Repository,
+
+    /// Loaded modification times for files in Git
+    mod_times: GitModificationTimes,
+}
+
+impl OsvExporter {
+    /// Load the the database at the given path
+    pub fn new(repo_path: Option<&Path>) -> Result<Self, Error> {
+        let repository = match repo_path {
+            Some(path) => Repository::open(path)?,
+            None => Repository::fetch_default_repo()?,
+        };
+        let mod_times = GitModificationTimes::new(&repository)?;
+        Ok(Self {
+            repository,
+            mod_times,
+        })
+    }
+
+    /// Exports all advisories to OSV JSON format to the specified directory.
+    pub fn export_all(&self, destination_folder: &Path) -> Result<(), Error> {
+        let repo_path = self.repository.path();
+        let collection_path = repo_path.join(Collection::Crates.as_str());
+        let mut found_at_least_one_advisory = false;
+
+        if let Ok(collection_entry) = fs::read_dir(&collection_path) {
+            for dir_entry in collection_entry {
+                for advisory_entry in fs::read_dir(dir_entry?.path())? {
+                    found_at_least_one_advisory = true;
+
+                    // Load the RustSec advisory
+                    let advisory_path = advisory_entry?.path();
+                    let advisory = Advisory::load_file(&advisory_path)?;
+                    let id = advisory.id().clone();
+
+                    if let Some(kind) = &advisory.metadata.informational {
+                        match kind {
+                            // If not `Unmaintained` or `Unsound` or `Notice`, don't export it to OSV
+                            // to make the output format stable.
+                            // Adding new types should be accompanied by a version bump.
+                            Informational::Unmaintained => (),
+                            Informational::Unsound => (),
+                            Informational::Notice => (),
+                            _ => continue,
+                        }
```

Comment:
I'm kind of surprised `clippy` doesn't suggest grouping these in some way... either a `match` with them all `|`'d, a `matches!` macro, or an array with `contains?`.

It's fine I guess, it just seems a bit more verbose than it needs to be.

Reply:
I tried using `|`, but `rustfmt` makes it look weird.

I took this over more complicated constructs because this is simple and stupid, and so easy to understand. I try to avoid clever code in order to make code easier to maintain, and immediately understandable even to someone not intimately familiar with Rust.